### PR TITLE
[xml] Missing group attributes on enum/bitfield arguments of core functions

### PR DESCRIPTION
**Pull request by SunSerega**
Part of #361

1. Some `glGet*` functions didn't have `group="GetPName"`.

2. `glShaderBinary`:
Specification defines at least one specific value - `GL_SHADER_BINARY_FORMAT_SPIR_V`.
So, i defined `ShaderBinaryFormat` group for it.\
\
Besides that - while looking for info i found quite a few people passing 0 as binary format. Is it some kind of special value "general format"? Should it also be added to `ShaderBinaryFormat` group?

3. `glFenceSync` and `glWaitSync` have `flags` parameter. It's reserved for now, so only 0 can be passed into them.
I added  `EmptyFlags` group, so that this restriction can be easier enforced in higl-level bindings.

I also applied all the same changes to extension functions, that define these functions as `<alias>`.

---

Somewhat off topic but related questions:

- `glSampleMaski` takes `GLbitfield` argument. And, well, it is a bitfield, but it's not what `GLbitfield` is usually used for.
Maybe it would make more sense to change that argument to `GLuint`, so it's not confused with actual enum arguments?

- I found and added few missing `group="sync"` attributes's in extension functions, alias to core functions that have this attribute.
But can someone actually explain why most arguments of type `GLsync` also have group `sync`?
Is it an old thing, from when there wasn't a `GLsync` type? Does it still need to exist?

### CODE REVIEW

File: xml/gl.xml [hunk 1]
```diff
@@ -5234,7 +5234,7 @@ typedef unsigned int GLhandleARB;
         <enum value="0" name="GL_FALSE" group="Boolean,VertexShaderWriteMaskEXT,ClampColorModeARB"/>
         <enum value="0" name="GL_NO_ERROR" group="GraphicsResetStatus,ErrorCode"/>
         <enum value="0" name="GL_ZERO" group="TextureSwizzle,StencilOp,BlendingFactor"/>
-        <enum value="0" name="GL_NONE" group="TextureCompareMode,PathColorFormat,CombinerBiasNV,CombinerScaleNV,DrawBufferMode,PixelTexGenMode,ReadBufferMode,ColorBuffer,PathGenMode,PathTransformType,PathFontStyle"/>
+        <enum value="0" name="GL_NONE" group="EmptyFlags,TextureCompareMode,PathColorFormat,CombinerBiasNV,CombinerScaleNV,DrawBufferMode,PixelTexGenMode,ReadBufferMode,ColorBuffer,PathGenMode,PathTransformType,PathFontStyle"/>
```
Comment (Perksey) [inline xml/gl.xml:5237]: I wouldn't call this `EmptyFlags` as it's not future-proof. Perhaps `SyncBehaviorFlags`? (Yes, I know that there's no currently defined flags for the glFenceSync flags parameter, but maybe one day there will be...)

**Review by Perksey — CHANGES_REQUESTED** [review]
Thank you so much for this PR! Most of the groups look good, however...

File: xml/gl.xml (hunk 1 above)
Comment (SunSerega) [reply xml/gl.xml:5237]: Well, this only part of core functions, i imagine there would be more of empty enum/bitfield arguments in extensions.

They probably not getting any enum values any time soon, so it makes sense to use 2 common groups in all functions - `EmptyEnum` and `EmptyFlags`, to not clog up xml files. And then change group name if these functions do get enum values.

It shouldn't be any problem to change group name later, right?

Comment (Perksey) [reply xml/gl.xml:5237]: I really think we should keep GLenum and GLbitfield parameters without associated enumerants untagged, Empty* is a smell and a little pointless. Group changes later down the line are technically breaking if concerning the function signatures.

Comment (SunSerega) [reply xml/gl.xml:5237]: > Group changes later down the line are technically breaking if concerning the function signatures.

Well, adding missing `group="GetPName"` would also break signatures then...
But i guess changes that break signatures can at least be minimized.

**Review by Perksey — APPROVED** [review]

**Comment by pdaniell-nv** [comment]
@oddhack this is approved to merge assuming you're happy with it.

**Pull request merged**